src/libexpr/primops.cc

```diff
@@ -2439,6 +2440,27 @@ static RegisterPrimOp primop_unsafeGetAttrPos(PrimOp {
     .fun = prim_unsafeGetAttrPos,
 });
 
+void prim_unsafeGetLambdaDoc(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    state.forceFunction(*args[0], pos, "while evaluating the first argument to builtins.unsafeGetLambdaDoc");
+
+    Pos funPos = state.positions[args[0]->lambda.fun->pos];
+
+    Comment::Doc doc = Comment::lookupDoc(funPos);
+
+    if(doc.comment.empty()) {
+        v.mkNull();
+    } else {
+        v.mkString(doc.comment);
+    }
+}
+
+static RegisterPrimOp primop_unsafeGetLambdaDoc(PrimOp {
+    .name = "__unsafeGetLambdaDoc",
+    .arity = 1,
+    .fun = prim_unsafeGetLambdaDoc,
+});
+
```

Comment:
This needs to be behind an `--experimental` flag to prevent it from being used in nixpkgs.